src/plugins/discover/server/locator/searchsource_from_locator.ts

```diff
@@ -147,7 +148,17 @@ export function searchSourceFromLocatorFactory(services: LocatorServicesDeps) {
 
     // Inject sort
     if (savedSearch.sort) {
-      const sort = getSortForSearchSource(savedSearch.sort as Array<[string, string]>, index);
+      const defaultSortingSetting = await services.uiSettings.get(SORT_DEFAULT_ORDER_SETTING);
+      const uiSettingsSyncReplacement = {
+        get: (key: string) =>
+          key === SORT_DEFAULT_ORDER_SETTING ? defaultSortingSetting : undefined,
+      };
+
+      const sort = getSortForSearchSource({
+        sort: savedSearch.sort as Array<[string, string]>,
+        dataView: index,
+        uiSettings: uiSettingsSyncReplacement,
+      });
```

Comment:
@davismcphee 

> What's this code doing and does it change the behaviour of reporting at all?

It now reads `SORT_DEFAULT_ORDER_SETTING` setting for the default fallback case. Before is was implicitly using `desc`. 

Would you suggest to revert?

cc @kertal

Reply:
For me it sounds like this fixes and unknown issue 👍

Reply:
Thanks for the info! I agree, sounds like a fix and makes sense to leave it to me.

But just in case, @tsullivan could you take a quick look at the change to see if it could impact anything on the reporting side?

Reply:
I don't see anything that would impact the `csv_v2` reporting export type, which is what relies on this code. However, from the code it looks like there could be more clarity to pass `defaultDirection` to `getSortForSearchSource`, rather than passing the wrapper called `uiSettingsSyncReplacement`. By design, the wrapper only returns the default direction or undefined, but `getSortForSearchSource` isn't aware of that possibly important information.

Reply:
@tsullivan Thanks, good point! Updated via https://github.com/elastic/kibana/pull/163784/commits/37a609522ece6d33b75bb527321628a5a1b01616